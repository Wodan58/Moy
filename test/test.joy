DEFINE
use-newton == [ [ .- ] cons ] dip swoncat 1.0 swap newton ;
swoncat == swap concat ;
newton == dup deriv [ 2 pick 2 pick i fabs 0.001 .> ] [ [ dupd ] dip
dup2 [ [ cleave ./ .- ] dip ] dip ] while pop pop ;
dup2 == dupd dup swapd ;
deriv == [ unary2 swap .- 0.001 ./ ] cons [ dup 0.001 .+ ] swoncat ;
cube-root == [ dup dup .* .* ] use-newton ;
put_list_str == "[ " putchars [ put_str 32 putch ] step "]" putchars ;
nl == '\n putch ;
END

1001.1 cube-root put_dbl nl .
argc .
argv put_list_str nl .
quit .
