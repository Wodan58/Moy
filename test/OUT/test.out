10.0037 
1 
["test.joy"] 
